Extensions/xkit_updates.js

```diff
@@ -292,7 +292,7 @@ XKit.extensions.xkit_updates = new Object({
 				try {
 					var mdata = JSON.parse(response.responseText);
 
-					if (mdata.malicious === true || mdata.malicious == "true") {
+					if (mdata.malicious || mdata.malicious == "true") {
```

Comment:
Hm, did not know that. I did it on multiple occasions, so I'll have to go
through and find them all.

On Wed, Aug 12, 2015 at 11:08 PM Evan <notifications@github.com> wrote:

> In Extensions/xkit_updates.js
> <https://github.com/new-xkit/XKit/pull/220#discussion_r36914065>:
>
> > @@ -292,7 +292,7 @@ XKit.extensions.xkit_updates = new Object({
> >  				try {
> >  					var mdata = JSON.parse(response.responseText);
> >
> > -					if (mdata.malicious === true || mdata.malicious == "true") {
> > +					if (mdata.malicious || mdata.malicious == "true") {
>
> @ThePsionic <https://github.com/ThePsionic> this is not how javascript
> works. if(x) coerces the value to a boolean, and therefore if m.malicious
> was "false" it would coerce to boolean as true (because it is a non-empty
> string)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/new-xkit/XKit/pull/220/files#r36914065>.
>